src/cli.rs

```diff
@@ -1,88 +1,156 @@
-                        .help("Remove oldest artifacts until the target directory is below the specified size in MB")
-                        .takes_value(true),
-                )
-                .arg(
-                    Arg::with_name("time")
-                        .short("t")
-                        .long("time")
-                        .value_name("days")
-                        .help("Number of days backwards to keep")
-                        .takes_value(true),
-                )
-                .group(
-                    ArgGroup::with_name("timestamp")
-                        .args(&["stamp", "file", "time", "installed", "toolchains", "maxsize"])
-                        .required(true),
-                )
-                .arg(
-                    Arg::with_name("path")
-                        .index(1)
-                        .value_name("path")
-                        .help("Path to check"),
-                ),
-        )
-        .get_matches()
+use clap::{ArgGroup, Parser};
+use std::path::PathBuf;
+
+pub fn parse() -> Args {
+    SweepArgs::parse().into_args()
+}
+
+#[derive(Parser)]
+#[command(version, about)]
+pub struct SweepArgs {
+    #[command(subcommand)]
+    sweep: SweepCommand,
+}
+
+impl SweepArgs {
+    fn into_args(self) -> Args {
+        let SweepCommand::Sweep(args) = self.sweep;
+        args
+    }
+}
+
+#[derive(clap::Subcommand)]
+pub enum SweepCommand {
+    Sweep(Args),
+}
+
+#[derive(Parser, Debug)]
+#[cfg_attr(test, derive(Default, PartialEq))]
+#[command(group(
+    ArgGroup::new("criterion")
+        .required(true)
+        .args(["stamp", "file", "time", "installed", "toolchains", "maxsize"])
+))]
+pub struct Args {
+    /// Path to check
+    pub path: Option<PathBuf>,
+
+    /// Dry run which will not delete any files
+    #[arg(short, long)]
+    pub dry_run: bool,
+
+    /// Load timestamp file in the given path, cleaning everything older
+    #[arg(short, long)]
+    file: bool,
+
+    #[arg(
+        long,
+        help = "The `recursive` flag defaults to ignoring directories \
+            that start with a `.`, `.git` for example is unlikely to include a \
+            Cargo project, this flag changes it to look in them"
+    )]
+    pub hidden: bool,
+
+    /// Keep only artifacts made by Toolchains currently installed by rustup
+    #[arg(short, long)]
+    installed: bool,
+
+    /// Remove oldest artifacts until the target directory is below the specified size in MB
+    #[arg(short, long, value_name = "MAXSIZE_MB")]
+    maxsize: Option<u64>,
+
+    /// Apply on all projects below the given path
+    #[arg(short, long)]
+    pub recursive: bool,
+
+    /// Store timestamp file at the given path, is used by file option
+    #[arg(short, long)]
+    stamp: bool,
+
+    /// Number of days backwards to keep
+    #[arg(short, long)]
+    time: Option<u64>,
+
+    /// Toolchains (currently installed by rustup) that should have their artifacts kept
```

Comment:
```suggestion
    /// Toolchains currently installed by rustup that should have their artifacts kept (comma separated)

Reply:
Feel free to review the DocComments, cause I just copied from the previous one.